[CI] RestTableTests testPlainTextChunking failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/e3qcjgx4i7t56/tests/:server:test/org.elasticsearch.rest.action.cat.RestTableTests/testPlainTextChunking

**Reproduction line:**
```
./gradlew ':server:test' --tests "org.elasticsearch.rest.action.cat.RestTableTests.testPlainTextChunking" -Dtests.seed=DA3B79BB209B6B3C -Dtests.locale=lv -Dtests.timezone=Asia/Kashgar -Druntime.java=17
```

**Applicable branches:**
main

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.rest.action.cat.RestTableTests&tests.test=testPlainTextChunking

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: (a value greater than <1> and <53>)
     but: <53> was <52>

  at __randomizedtesting.SeedInfo.seed([DA3B79BB209B6B3C:AFCBA1B446FA953A]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.rest.action.cat.RestTableTests.testPlainTextChunking(RestTableTests.java:292)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

```

--- a/server/src/test/java/org/elasticsearch/rest/action/cat/RestTableTests.java
+++ b/server/src/test/java/org/elasticsearch/rest/action/cat/RestTableTests.java
@@ -11,6 +11,7 @@ package org.elasticsearch.rest.action.cat;
 import org.apache.lucene.util.BytesRef;
 import org.elasticsearch.common.Table;
 import org.elasticsearch.common.recycler.Recycler;
+import org.elasticsearch.common.unit.ByteSizeUnit;
 import org.elasticsearch.common.xcontent.XContentHelper;
 import org.elasticsearch.rest.AbstractRestChannel;
 import org.elasticsearch.rest.RestResponse;
@@ -29,7 +30,6 @@ import java.util.concurrent.atomic.AtomicInteger;
 
 import static org.elasticsearch.rest.action.cat.RestTable.buildDisplayHeaders;
 import static org.elasticsearch.rest.action.cat.RestTable.buildResponse;
-import static org.hamcrest.Matchers.allOf;
 import static org.hamcrest.Matchers.contains;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.equalToIgnoringCase;
@@ -259,22 +259,28 @@ public class RestTableTests extends ESTestCase {
         assertEquals(Arrays.asList(1, 0, 2), rowOrder);
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/92181")
     public void testPlainTextChunking() throws Exception {
+        final var cells = randomArray(8, 8, String[]::new, () -> randomAlphaOfLengthBetween(1, 5));
+        final var expectedRow = String.join(" ", cells) + "\n";
+
+        // OutputStreamWriter has an 8kiB buffer so all chunks are at least that big
+        final var bufferSize = ByteSizeUnit.KB.toIntBytes(8);
+        final var rowLength = expectedRow.length();
+        final var expectedRowsPerChunk = 1 + bufferSize / rowLength; // end chunk after first row which overflows the buffer
+        final var expectedChunkSize = expectedRowsPerChunk * rowLength;
+
+        final var rowCount = between(expectedRowsPerChunk + 1, expectedRowsPerChunk * 10);
+        final var expectedChunkCount = 1 + (rowCount * rowLength - 1) / expectedChunkSize; // ceil(rowCount * rowLength / expectedChunkSize)
+        assertThat(expectedChunkCount, greaterThan(1));
+
         final var expectedBody = new StringBuilder();
-        final var rowCount = between(10000, 20000);
         for (int i = 0; i < rowCount; i++) {
             table.startRow();
-            table.addCell("foo");
-            table.addCell("foo");
-            table.addCell("foo");
-            table.addCell("foo");
-            table.addCell("foo");
-            table.addCell("foo");
-            table.addCell("foo");
-            table.addCell("foo");
+            for (final var cell : cells) {
+                table.addCell(cell);
+            }
             table.endRow();
-            expectedBody.append(TEXT_TABLE_BODY);
+            expectedBody.append(expectedRow);
         }
 
         final var request = new FakeRestRequest.Builder(xContentRegistry()).withHeaders(
@@ -286,13 +292,10 @@ public class RestTableTests extends ESTestCase {
             public void sendResponse(RestResponse response) {}
         });
 
-        // OutputStreamWriter has an 8kiB buffer so all chunks are at least that big anyway
-        final var bodyChunks = getBodyChunks(response, 8192);
-        final var rowLength = TEXT_TABLE_BODY.length();
-        final var expectedChunkSize = ((8193 + rowLength) / rowLength) * rowLength; // ceil(8193/rowLength) * rowLength
-        assertThat(bodyChunks.size(), allOf(greaterThan(1), equalTo((rowCount * rowLength + expectedChunkSize) / expectedChunkSize)));
-        assertThat(bodyChunks.get(0).length(), equalTo(expectedChunkSize));
-        assertEquals(expectedBody.toString(), String.join("", bodyChunks));
+        final var bodyChunks = getBodyChunks(response, bufferSize);
+        assertEquals("chunk count", expectedChunkCount, bodyChunks.size());
+        assertEquals("first chunk size", expectedChunkSize, bodyChunks.get(0).length());
+        assertEquals("body contents", expectedBody.toString(), String.join("", bodyChunks));
     }
 
     public void testEmptyTable() throws Exception {

